google/cloud/forseti/enforcer/gce_firewall_enforcer.py

```diff
@@ -923,16 +933,44 @@ def _update_rules(self, network):
                 self.expected_rules.rules[rule_name]
                 for rule_name in self._rules_to_update
             ], network)
-            patch_function = self.compute_client.patch_firewall_rule
-            (successes, failures, change_errors) = self._apply_change(
-                patch_function, rules)
-            self._updated_rules.extend(successes)
-            change_count += len(successes)
-            if failures:
-                raise FirewallEnforcementUpdateFailedError(
-                    'Firewall enforcement failed while deleting rules for '
-                    'project {}. The following errors were encountered: {}'
+            rules_to_patch = []
+            rules_to_replace = []
+            for rule in rules:
+                if _rule_update_can_patch(self.current_rules.rules[rule['name']], rule):
```

Comment:
This is checking if `allowed` or `denied is in the rule name. Instead should it be checking if the rule has a non-empty `allowed` or `denied` list ([firewall api ref](https://cloud.google.com/compute/docs/reference/rest/v1/firewalls))?